### PR TITLE
Update CI config for Flutter 2

### PR DESCRIPTION
**Pull request by stuartmorgan**
Includes cleanup to simplify our setup. Major changes:
- Eliminate the NNBD plugin filtering for stable.
- Remove the temporarily-added beta branch testing.
- Enable Linux, macOS, and web on stable (Windows is LUCI-based)
- Combine the two different macOS matrix configurations now that they
  are the same.
- Combine the two different Linux matrix configurations by using a single
  Dockerfile (which now also includes clang-format)
  - The web integration smoke test temporarily still uses the old Dockerfile,
    now renamed, because the driver installer script doesn't support
    Chrome 89 yet.
- Move most of the Linux tasks to lower-CPU machines to allow more
  tasks to run in parallel without hitting the community limit.
- Reorder the tasks slightly and give them comments to identify
  platform groupings
- Enabled web "build all plugins together" and "build all examples"
  tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
Looks like there are issues with regenerating the Docker instances that would need investigation and reworking, so reverting the merging of the Dockerfiles.

File: .cirrus.yml [hunk 1]
```diff
@@ -77,6 +121,27 @@ task:
       test_script:
         - cd $INTEGRATION_TEST_PATH/example/
         - flutter drive -v --driver=test_driver/integration_test.dart --target=integration_test/example_test.dart -d web-server --release --browser-name=chrome
+
+# Heavy-workload tasks.
+# These use machines with more CPUs and memory, so will reduce parallelization
+# for non-credit runs.
+task:
```
Comment (blasten) [inline .cirrus.yml:128]: does it need `cpu` and `memory` values under `container`?

Comment (stuartmorgan) [reply .cirrus.yml:128]: No, there are defaults (2 CPUs and 4 GB)

Comment (blasten) [reply .cirrus.yml:128]: I was wondering what makes these tasks use machines with more CPUs and memory as mentioned in the comment above?

Comment (stuartmorgan) [reply .cirrus.yml:128]: 🤦 

I didn't look carefully enough at the context and thought this was about the light-load config. Fix incoming.

**Review by blasten — APPROVED** [review]
LGTM

**Pull request merged**